.github/workflows/ci.yml

```diff
@@ -7,6 +7,38 @@ on:
       - master
 
 jobs:
+  check-formatting:
+    name: 'Check Formatting of Changes'
+    runs-on: windows-latest
+    steps:
+      - name: Check Formatting
+        run: |
+          if NOT EXIST "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat" (
```

Comment:
Unsure.  This is the approach used by WIL, which seems to have been stable for at least a year without any breakage that I've seen.

Reply:
Just to comment for WIL, AFAIK running on `windows-2022` ensures VS 2022. IIRC I actually copied that portion of the CI yaml from C++/WinRT long ago. But it looks like other parts of the pipeline have:
```
	          $VSDevCmd = & "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere" -latest -find Common7\tools\VSDevCmd.bat
          if (!$VSDevCmd) { exit 1 }	          if (!$VSDevCmd) { exit 1 }
          echo "Using VSDevCmd: ${VSDevCmd}"	          echo "Using VSDevCmd: ${VSDevCmd}"
          Add-Content $env:GITHUB_ENV "VSDevCmd=$VSDevCmd"	          Add-Content $env:GITHUB_ENV "VSDevCmd=$VSDevCmd"
```

Reply:
WindowsAppSDK uses [vswhere ](https://github.com/microsoft/vswhere)to look for the environment variable `$VCToolsInstallDir` so that it can then run `VsDevCmd.bat` to obtain a developer command prompt.

Reply:
Updated to use vswhere.